server/src/lib/config.rs

```diff
@@ -69,11 +82,43 @@ impl Device {
     }
 }
 
+#[derive(Deserialize, Debug)]
+pub struct BlockDevice {
+    #[serde(default, rename = "type")]
+    pub bdtype: String,
+
+    #[serde(flatten, default)]
+    pub options: BTreeMap<String, toml::Value>,
+}
+
+impl BlockDevice {
+    pub fn block_dev<R: propolis::block::BlockReq>(
```

Comment:
IMO it's weird to have a function named "block device" on a structure named "block device" - it's not intuitive what this does based on name.

Can we call this `create_block_device`, perhaps?

Reply:
agreed, fixed in e3b4ea9